app/services/model_service.py

```diff
@@ -74,3 +99,24 @@ def _prepare_dataset(
         elif model_class == "logistic":
             y = df["G3"] >= 15.0
         return X, y
+    
+    @staticmethod
+    def _save_model_metadata(model_id: str, model_metadata: ModelMetadata) -> None:
+        filepath = f"../data/models/model_metadata/{model_metadata.model_class}_{model_id}.txt"
```

Comment:
I think it would be safer to use `pathlib` to build the file path.

ex. 
```python
data_dir = Path().cwd().parent.joinpath("data")
filepath = data_dir.joinpath(f"models/model_metadata/{model_id}.pkl"))
```

Reply:
Fixed!